We were working with music reviews dataset. The goal was to estimate the confidence interval for the mean of reviews scores.

In [1]:
from scipy import stats
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
%matplotlib inline

In [4]:
#import necessary modules

import psycopg2
import pandas as pd

#create connection
conn = psycopg2.connect('dbname=pitchfork_reviews')
cur = conn.cursor()

#execute simple commands
cur.execute("""
SELECT score FROM reviews;
""")

scores = cur.fetchall()
df_scores = pd.DataFrame(scores, columns=['score'])

In [11]:
scores = np.array(df_scores.score)

In [12]:
scores

array([9.3, 7.9, 7.3, ..., 6.3, 7.2, 6.5])

In [13]:
sample = np.random.choice(scores, size=1000)

In [15]:
bootstrap_samples = []
bootstrap_sample_means = np.zeros(1000)

for i in range(1000):
    bootstrap_sample = np.random.choice(sample, size=1000)
    bootstrap_samples.append(bootstrap_sample)
    bootstrap_sample_means[i] = bootstrap_sample.mean()
 

In [18]:
np.percentile(a=bootstrap_sample_means, q=2.5)

6.9415925

In [19]:
np.percentile(a=bootstrap_sample_means, q=97.5)

7.105202500000001

### Conclusion
The confidence interval for the mean score is 6.94 to 7.10 with confidence level of 95%. That means that 95 times out of 100 the mean will fall into this interval.